lib/index.spec.js

```diff
@@ -163,45 +180,201 @@ describe('AssociateWafPlugin', () => {
     })
-  })
 
-  describe('with configuration', () => {
-    beforeEach(() => {
-      serverless.service.custom = {
-        associateWaf: {
-          name: 'stage-service-name'
+    describe('with invalid property "custom.associateWaf.version"', () => {
+      beforeEach(() => {
+        serverless.service.custom = {
+          associateWaf: {
+            version: "invalid-version"
+          }
         }
-      }
-      serverless.service.provider.compiledCloudFormationTemplate = {
-        Outputs: [
-          'ApiGatewayRestApiWaf', {Description: 'Rest API Id', Value: 'some-api-rest-id'}]
-      }
-      plugin = new AssociateWafPlugin(serverless, options)
-    })
+        plugin = new AssociateWafPlugin(serverless, options)
+      })
+
+      it('should set hooks if ', () => {
+        expect(plugin.hooks).toHaveProperty('after:deploy:deploy')
+        expect(plugin.hooks).toHaveProperty('before:package:finalize')
+      })
+
+      it('should log info when invalid version configuration exists ', () => {
+        expect(plugin.serverless.cli.log).toHaveBeenLastCalledWith(expect.stringContaining('Invalid WAF Version Configuration'))
+      })
+
+      it(`wafVersion should equal ${WAF_REGIONAL}`, () => {
+        expect(WAF_REGIONAL).toEqual(plugin.wafVersion);
+      })
+
+
+      it('"outputRestApiId()" should be invoked', async () => {
+        await expectOutputRestApiIdToHaveBeenCalled(plugin)
+      })
 
-    it('should set config', () => {
-      expect(plugin.config).toBeTruthy()
+      it('"disassociateWaf()" should be invoked', async () => {
+        await expectDisassociateWafToHaveBeenCalled(plugin)
+      })
     })
 
-    it('should set hooks', () => {
-      expect(plugin.hooks).toHaveProperty('after:deploy:deploy')
-      expect(plugin.hooks).toHaveProperty('before:package:finalize')
+    describe('with null property "custom.associateWaf.version"', () => {
+      beforeEach(() => {
+        serverless.service.custom = {
+          associateWaf: {}
+        }
+        serverless.service.provider.compiledCloudFormationTemplate = {
+          Outputs: [
+            {}]
+        }
+        plugin = new AssociateWafPlugin(serverless, options)
+      })
+
+      it('should set hooks if ', () => {
+        expect(plugin.hooks).toHaveProperty('after:deploy:deploy')
+        expect(plugin.hooks).toHaveProperty('before:package:finalize')
+      })
+
+      it(`wafVersion should equal ${WAF_REGIONAL}`, () => {
+        expect(WAF_REGIONAL).toEqual(plugin.wafVersion);
+      })
+
+      it('"outputRestApiId()" should be invoked', async () => {
+        await expectOutputRestApiIdToHaveBeenCalled(plugin)
+      })
+
+      it('"disassociateWaf()" should be invoked', async () => {
+        await expectDisassociateWafToHaveBeenCalled(plugin)
+      })
     })
 
-    it('"outputRestApiId()" should be invoked', async () => {
-      await expectOutputRestApiIdToHaveBeenCalled(plugin)
+  })
+
+  describe('with configuration', () => {
```

Comment:
Just a tip for the future, try to minimize the noise from a diff so that existing unchanged blocks of code aren't mistaken as being "removed" and "added". Reading through the diff of `index.spec.js` was quite noisy and made it way more difficult to easily review what code you added/ modified.

Reply:
Thanks, thats great feedback. I found most of the reason for the noise due to autoformatting from vscode. Will make sure to turn that off in the future for stuff like this.